Sources/ServiceDiscovery/ServiceDiscovery.swift

```diff
@@ -12,179 +12,26 @@
 //
 //===----------------------------------------------------------------------===//
 
-import Atomics
-import Dispatch
-
 // MARK: - Service discovery protocol
 
 /// Provides service instances lookup.
 ///
 /// ### Threading
 ///
 /// `ServiceDiscovery` implementations **MUST be thread-safe**.
-public protocol ServiceDiscovery: AnyObject {
-    /// Service identity type
-    associatedtype Service: Hashable
-    /// Service instance type
-    associatedtype Instance: Hashable
-
-    /// Default timeout for lookup.
-    var defaultLookupTimeout: DispatchTimeInterval { get }
+public protocol ServiceDiscovery: Sendable {
+    /// Service discovery instance type
+    associatedtype Instance: Sendable
+    /// AsyncSequence of Service discovery instances
+    associatedtype DiscoverySequence: AsyncSequence where DiscoverySequence.Element == [Instance]
```

Comment:
Do we want to require that the `DiscoverySequence` is `Sendable`? This will make usage sites much simpler.